drivers/soundwire/intel.c

```diff
@@ -1976,17 +1982,24 @@ static const struct dev_pm_ops intel_pm = {
 	SET_RUNTIME_PM_OPS(intel_suspend_runtime, intel_resume_runtime, NULL)
 };
 
-static struct platform_driver sdw_intel_drv = {
-	.probe = intel_master_probe,
-	.remove = intel_master_remove,
+static const struct ancillary_device_id intel_link_id_table[] = {
+	{ .name = "soundwire_intel.intel-sdw" },
```

Comment:
hm, I guess "intel-sdw" is required for matching, right? But as is it seems twice redundant to me... Wouldn't just "intel-sdw" work? Or something...

Reply:
yes, thats a nice catch. The first part is the module name that gets appended to the device name by the ancillary bus. So maybe something lile soundwire_intel.link instead might be better?

Reply:
@ranj063 'intel-sdw" is currently the name for the driver. Are you saying the ID table and driver names are unrelated? I thought the ID stable had to be KBUILD_MOD.driver_name?

Reply:
@plbossart no the first part is KBUILD_MODULENAME and the second part is the name you choose for the ancillary device. It can be anything really.

Reply:
fixed, the device name is now soundwire_intel.link.0..3
much better than before.